File: webapp-net/AudienceManager/Models/LoginForm.cs
using Sdl.Web.Common.Models;
using System.ComponentModel.DataAnnotations;

namespace Sdl.Web.Modules.AudienceManager.Models
{
    /// <summary>
    /// LoginForm entity
    /// </summary>
    public class LoginForm : EntityModel
    {
        /// <summary>
        /// Holds the form control value for username
        /// </summary>
        [Required(ErrorMessage = "@Model.NoUserNameMessage")]
        [SemanticProperty(IgnoreMapping=true)]
        public string UserName { get; set; }

        /// <summary>
        /// Holds the form control value for password
        /// </summary>
        [Required(ErrorMessage = "@Model.NoPasswordMessage")]
        [SemanticProperty(IgnoreMapping = true)]
        public string Password { get; set; }

        /// <summary>
        /// Holds the form control value for remember me
        /// </summary>
        [SemanticProperty(IgnoreMapping = true)]
        public bool RememberMe { get; set; }

        /// <summary>
        /// Form heading text
        /// </summary>
        public string Heading { get; set; }

        /// <summary>
        /// Label text for username input control on view
        /// </summary>
        public string UserNameLabel { get; set; }

        /// <summary>
        /// Label text for password input control on view
        /// </summary>
        public string PasswordLabel { get; set; }

        /// <summary>
        /// Label text for remember me check box on view
        /// </summary>
        public string RememberMeLabel { get; set; }

        /// <summary>
        /// Label text for submit botton on view
        /// </summary>
        public string SubmitButtonLabel { get; set; }

        /// <summary>
        /// User name not specified message
        /// </summary>
        public string NoUserNameMessage { get; set; }

        /// <summary>
        /// Password not specified message
        /// </summary>
        public string NoPasswordMessage { get; set; }

        /// <summary>
        /// Authentication error message
        /// </summary>
        public string AuthenticationErrorMessage { get; set; }
    }
}

File: webapp-java/dxa-module-audience-manager-abstract/src/main/java/com/sdl/dxa/modules/audience/model/LoginForm.java
package com.sdl.dxa.modules.audience.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticMappingIgnore;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.validation.BindingResult;

@Data
@EqualsAndHashCode(callSuper = true)
public class LoginForm extends AbstractEntityModel {

    @SemanticMappingIgnore
    @JsonIgnore
    private String userName;

    @SemanticMappingIgnore
    @JsonIgnore
    private String password;

    @SemanticMappingIgnore
    @JsonIgnore
    private boolean rememberMe;

    @SemanticMappingIgnore
    @JsonIgnore
    private BindingResult bindingResult;

    private String heading;

    private String userNameLabel;

    private String passwordLabel;

    private String rememberMeLabel;

    private String submitButtonLabel;

    private String noUserNameMessage;

    private String noPasswordMessage;

    private String authenticationErrorMessage;
}